libraries/pbrlib/genglsl/mx_blackbody.glsl

```diff
@@ -0,0 +1,49 @@
+/// XYZ to Rec.709 RGB colorspace conversion
+const mat3 XYZ_to_RGB = mat3( 3.2406, -0.9689, 0.0557,
+                             -1.5372, 1.8758, -0.2040,
+                             -0.4986, 0.0415, 1.0570);
+
+void mx_blackbody(float temperatureKelvin, out vec3 colorValue)
+{
+    float xc, yc;
+    float t, t2, t3, xc2, xc3;
+
+    t = 1000.0 / temperatureKelvin;
+    t2 = t * t;
+    t3 = t * t * t;
+
+    // if value outside valid range of approximation clamp to accepted temperature range
+    clamp(temperatureKelvin, 1667.0, 25000.0);
```

Comment:
Should you not assign the result here? Like so:

`temperatureKelvin = clamp(temperatureKelvin, 1667.0, 25000.0);`